### ISSUE TITLE
Issue with Ground Truth Mapping: Class IDs from a Different Dataset Mapped to COCO Class Names

### ISSUE THREAD

**Issue by Agh404180** [ultralytics/ultralytics#19796]
### Search before asking

- [ ] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

**When validating a model trained on the COCO dataset using a different dataset, the ground truth bounding box coordinates and class IDs are taken from the label file of the new dataset. However, the class IDs are being mapped to the class names of the COCO dataset instead of the class names from the new dataset. Why is this happening?**

Dataset used : https://github.com/ultralytics/ultralytics/blob/main/ultralytics/cfg/datasets/medical-pills.yaml

This is the code I have run : 

!pip install ultralytics 

from ultralytics import YOLO

# Load YOLOv8 pose model
model = YOLO("yolov8n.pt")

# Validate on the hand keypoints dataset
metrics = model.val(data="/content/medical-pills.yaml", save_json=True)

# Print results
print(metrics)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Agh404180, thank you for your interest in Ultralytics 🚀! We recommend checking out the [Docs](https://docs.ultralytics.com/) for guidance, where you'll find [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples, as well as answers to many common questions.

If this is a 🐛 Bug Report, could you please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us investigate the issue further? This includes your script, dataset structure, and any relevant logs or outputs.

For ❓ Questions related to custom training, ensure you’re following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) and share additional details like dataset examples and training configurations.

Join the Ultralytics community to connect with others and get more support:
- For real-time discussions, visit [Discord](https://discord.com/invite/ultralytics) 🎧
- For in-depth topics, check out [Discourse](https://community.ultralytics.com/)
- Or join our [Subreddit](https://reddit.com/r/Ultralytics) to share ideas!

## Upgrade

Please ensure you are using the latest `ultralytics` package. Update it along with its [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

You can run YOLO in any of the following verified environments (all with required dependencies preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests validate YOLO functionality across tasks and modes on macOS, Windows, and Ubuntu daily.

This is an automated response 🛠️, but an Ultralytics engineer will review and assist you shortly. Thank you for your patience! 😊

**Comment by Y-T-G** [comment]
Because you're using a model trained on COCO dataset.

**Comment by Agh404180** [comment]
Im using a pretrained model(yolov8 (ultralytics)) which is trained on COCO dataset.(COCO has 80 classes). Now I am validating it on medical-pills dataset which has only 1 class.

After running this code,

from ultralytics import YOLO

# Load YOLOv8 pose model
model = YOLO("yolov8n.pt")

# Validate on the hand keypoints dataset
metrics = model.val(data="/content/medical-pills.yaml", save_json=True)

# Print results
print(metrics)

Im getting all evaluation metrics and also val_batch0_labels.jpg and val_batch0_pred.jpg

In val_batch0_labels.jpg i am getting pills images all with bounding boxes detected as person(which is the first class in COCO) 

and in val_batch0_pred.jpg
i am getting pills images as cup,cake,dining table,bottle(which all are classes of COCO dataset)

Why its taking pill as person,instead of taking it as pill.

Is there any class mismatch? Or is the code taking yaml file of COCO?
